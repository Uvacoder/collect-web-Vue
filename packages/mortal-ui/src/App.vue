<script setup lang="ts">
import { LoadingSquare } from '@/components'
import { Reviewer } from '@/features'
</script>

<template>
  <suspense>
    <template #default>
      <Reviewer />
    </template>
    <template #fallback>
      <LoadingSquare />
    </template>
  </suspense>
</template>

<style scoped>
</style>
