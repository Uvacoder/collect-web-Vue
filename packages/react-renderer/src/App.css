.app-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .app-logo {
    animation: app-logo-spin infinite 20s linear;
  }
}

.app-header {
  @apply flex flex-col items-center justify-center;
  @apply bg-gray-900;

  min-height: 100vh;
  font-size: calc(10px + 2vmin);
}

@keyframes app-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
