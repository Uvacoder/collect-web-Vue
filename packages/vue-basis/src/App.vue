<template>
  <div id="nav">
    <router-link :to="{ name: 'EventList' }">
      Events
    </router-link> |
    <router-link :to="{ name: 'About' }">
      About
    </router-link>
  </div>
  <suspense>
    <template #default>
      <router-view />
    </template>
    <template #fallback>
      <div>Loading events list ...</div>
    </template>
  </suspense>
</template>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
  text-align: center;
}

#nav {
  padding: 30px;
}

#nav a {
  font-weight: bold;
  color: #2c3e50;
}

#nav a.router-link-exact-active {
  color: #42b983;
}

h4 {
  font-size: 20px;
}
</style>
