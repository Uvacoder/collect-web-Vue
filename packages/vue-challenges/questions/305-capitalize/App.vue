<script setup>
import { ref } from 'vue';

const value = ref('');
</script>

<template>
  <!-- eslint-disable-next-line vue/valid-v-model -->
  <input v-model.capitalize="value" type="text" />
</template>
