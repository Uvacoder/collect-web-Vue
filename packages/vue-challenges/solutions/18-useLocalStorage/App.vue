<script setup lang="ts">
import { customRef } from 'vue';

/**
 * Implement the composable function
 * Make sure the function works correctly
 */
function useLocalStorage(key: string, initialValue: any) {
  return customRef((track, trigger) => ({
    get() {
      track();
      return localStorage.getItem(key) || initialValue;
    },
    set(value) {
      localStorage.setItem(key, value);
      trigger();
    },
  }));
}

const counter = useLocalStorage('counter', 0);

// We can get localStorage by triggering the getter:
console.log(counter.value);

// And we can also set localStorage by triggering the setter:
const update = () => counter.value++;
</script>

<template>
  <p>Counter: {{ counter }}</p>
  <button @click="update">Update</button>
</template>
