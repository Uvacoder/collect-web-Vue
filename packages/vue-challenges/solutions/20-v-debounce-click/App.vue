<script setup lang="ts">
/**
 * Implement the custom directive
 * Make sure the `onClick` method only gets triggered once when clicked many times quickly
 * And you also need to support the debounce delay time option. e.g `v-debounce-click:ms`
 *
 */

const debounce = (fn: Function, delay: number) => {
  let timer = null;

  return function () {
    if (timer) return;
    fn();
    timer = setTimeout(() => (timer = clearTimeout(timer)), delay);
  };
};

const vDebounceClick = {
  mounted(el: HTMLElement, { value, arg }: { value: Function; arg: number }) {
    el.addEventListener('click', debounce(value, arg));
  },
};

function onClick() {
  console.log('Only triggered once when clicked many times quickly');
}
</script>

<template>
  <button v-debounce-click:200="onClick">Click on it many times quickly</button>
</template>
