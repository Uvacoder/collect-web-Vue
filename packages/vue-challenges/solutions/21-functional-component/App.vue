<script setup lang="ts">
import { ref, h } from 'vue';

interface ListProps {
  list: {
    name: string;
  }[];
  'active-index': number;
  onToggle(index: number): void;
}

/**
 * Implement a functional component :
 * 1. Render the list elements (ul/li) with the list data
 * 2. Change the list item text color to red when clicked.
 */
const ListComponent = (props: ListProps) => {
  return h(
    'ul',
    null,
    props.list.map((item, index) =>
      h(
        'li',
        {
          key: index,
          style:
            props['active-index'] === index
              ? {
                  color: 'red',
                }
              : undefined,
          onClick: props.onToggle.bind(null, index),
        },
        item.name
      )
    )
  );
};

const list = [
  {
    name: 'John',
  },
  {
    name: 'Doe',
  },
  {
    name: 'Smith',
  },
];

const activeIndex = ref(0);

function toggle(index: number) {
  activeIndex.value = index;
}
</script>

<template>
  <list-component :list="list" :active-index="activeIndex" @toggle="toggle" />
</template>
