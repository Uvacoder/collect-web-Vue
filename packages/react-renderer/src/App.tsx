import { useCallback, useState } from 'react';
import Logo from './logo.svg';
import './App.css';

function App() {
  const [count, setCount] = useState<number>(0);
  const inc = useCallback(() => {
    setCount(count => count + 1);
  }, []);
  const dec = useCallback(() => {
    setCount(count => count - 1);
  }, []);

  return (
    <div className="app">
      <header className="app-header">
        <Logo className="app-logo" title="logo" />
        <div className="flex flex-row justify-center items-center">
          <button className="p-4 text-white text-6xl" onClick={dec}>
            -
          </button>
          <div className="p-4 text-white text-6xl">{count}</div>
          <button className="p-4 text-white text-6xl" onClick={inc}>
            +
          </button>
        </div>
      </header>
    </div>
  );
}

export default App;
