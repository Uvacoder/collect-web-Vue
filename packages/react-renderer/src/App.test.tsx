import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('Renderer', () => {
  it('should render counter', () => {
    render(<App />);
    const incElement = screen.getByText(/\+/i);
    const decElement = screen.getByText(/-/i);
    const countElement = screen.getByText(0);

    expect(incElement).toBeInTheDocument();
    expect(decElement).toBeInTheDocument();
    expect(countElement).toBeInTheDocument();
    expect(countElement.textContent).toBe('0');
  });

  it('should increment counter', () => {
    render(<App />);
    const incElement = screen.getByText(/\+/i);
    const countElement = screen.getByText(0);

    fireEvent.click(incElement);
    expect(countElement.textContent).toBe('1');
    fireEvent.click(incElement);
    expect(countElement.textContent).toBe('2');
    fireEvent.click(incElement);
    expect(countElement.textContent).toBe('3');
  });

  it('should decrement counter', () => {
    render(<App />);
    const decElement = screen.getByText(/-/i);
    const countElement = screen.getByText(0);

    fireEvent.click(decElement);
    expect(countElement.textContent).toBe('-1');
    fireEvent.click(decElement);
    expect(countElement.textContent).toBe('-2');
    fireEvent.click(decElement);
    expect(countElement.textContent).toBe('-3');
  });

  it('should pass snapshot testing', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
