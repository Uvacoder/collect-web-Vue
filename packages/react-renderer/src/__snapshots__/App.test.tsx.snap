// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer should pass snapshot testing 1`] = `
<div>
  <div
    class="app"
  >
    <header
      class="app-header"
    >
      <svg
        class="app-logo"
        title="logo"
      />
      <div
        class="flex flex-row justify-center items-center"
      >
        <button
          class="p-4 text-white text-6xl"
        >
          -
        </button>
        <div
          class="p-4 text-white text-6xl"
        >
          0
        </div>
        <button
          class="p-4 text-white text-6xl"
        >
          +
        </button>
      </div>
    </header>
  </div>
</div>
`;
